src/Plugins/LevelDBStore/IO/Data/LevelDB/Iterator.cs

```diff
@@ -37,25 +40,44 @@ public void Dispose()
             }
         }
 
+        /// <summary>
+        /// Return the key for the current entry.
+        /// REQUIRES: Valid()
+        /// </summary>
         public byte[] Key()
         {
             IntPtr key = Native.leveldb_iter_key(handle, out UIntPtr length);
             CheckError();
             return key.ToByteArray(length);
         }
 
+        /// <summary>
+        /// Moves to the next entry in the source.
+        /// After this call, Valid() is true if the iterator was not positioned at the last entry in the source.
+        /// REQUIRES: Valid()
+        /// </summary>
         public void Next()
         {
             Native.leveldb_iter_next(handle);
             CheckError();
         }
 
+        /// <summary>
+        /// Moves to the previous entry in the source.
+        /// After this call, Valid() is true if the iterator was not positioned at the first entry in source.
+        /// REQUIRES: Valid()
```

Comment:
Does it? Can I go back to "c" in "a b c" after three "Next" calls and `!Valid()` (ended) iterator?

Reply:
Comemnt is from the @cschuchardt88 pr, i did not really check it, but anyway, if its not yet proper, i will remove it until the last minute.

Reply:
I'm just suggesting here, I don't know the real behavior, but we want comments to be correct wrt these details.

Reply:
This was a useful comment after all, the only question is `Valid()` requirement, please return it (maybe except for `Valid()` line if we don't know for sure). I'm also not sure `Valid()` requirement is appropriate for `Next()` (suppose `Prev()` at `a` of `a b c` and `Next()` then). Needs to be checked.